planner/scan.go

```diff
@@ -61,17 +63,84 @@ func (n *scanNode) Kind() string {
 
 func (n *scanNode) Init() error {
 	// init the fetcher
-	if err := n.fetcher.Init(&n.desc, n.fields, n.reverse, n.showDeleted); err != nil {
+	if err := n.fetcher.Init(&n.desc, n.fields, n.filter, &n.slct.DocumentMapping, n.reverse, n.showDeleted); err != nil {
 		return err
 	}
 	return n.initScan()
 }
 
 func (n *scanNode) initCollection(desc client.CollectionDescription) error {
 	n.desc = desc
+	n.initFields(n.slct.Fields)
```

Comment:
question: I dont understand why you are doing all this here still? If you have sorted everything out in the mapper, you should not have to do anything like this in the planner.

Reply:
Mentioned elsewhere briefly, but to summarize. All the changes in the `mapper` are to surface the fields we actually need/care about.

These fields are surfaced in the mapper, so they can be consumed here, by explicitly telling the fetcher (via the scanNode) which fields we care about.